### ISSUE TITLE
Compiler allows a global variable name to be the same as a function name

### ISSUE THREAD

**Issue by yzhang90** [ethereum/vyper#516]
### What's your issue about?
The compiler should rejects the following program:
```
foo: decimal[3] 
@public def foo(): 
    self.foo = [1, 2.1, 3]
```
because the global variable and the function have the same name and it is very confusing to write such program.

### How can it be fixed?

Checking name duplication in parser.py

#### Cute Animal Picture

![image](https://user-images.githubusercontent.com/5641590/33246550-5bf5ccf4-d2db-11e7-9c3e-c7bde400face.png)

**Comment by DavidKnott** [comment]
@yzhang90 On it 👍

**Comment by DavidKnott** [comment]
@fubuloubu I'm working on this.  Do you think the following should be allowed?
```
def foo():
    foo: num
```
I'm leaning towards yes, as I'm thinking that uniforming naming can increase readability.

**Comment by fubuloubu** [comment]
I'm not a big fan of that personally, but I think it would work because `foo()` is in the `self` "namespace" so that's fine

**Comment by jakerockland** [comment]
@yzhang90 @DavidKnott was poking around at bugs that could be addresses and ended up here, but it looks like this issue can be closed with #659 merged?

**Comment by jacqueswww** [comment]
@jakerockland that is correct ;) sorry for the later reply.
